src/VisualStudio/Core/Impl/CodeModel/NodeKeyValidation.cs

```diff
@@ -17,14 +17,17 @@ public NodeKeyValidation()
 
         public void AddProject(AbstractProject project)
         {
-            var provider = (IProjectCodeModelProvider)project;
-            var fcms = provider.ProjectCodeModel.GetCachedFileCodeModelInstances();
-
-            foreach (var fcm in fcms)
+            var provider = project as IProjectCodeModelProvider;
+            if (provider != null)
```

Comment:
What is the implication if the project doesn't implement `IProjectCodeModelProvider`?  Is it a bug that we should be logging?